lib/matplotlib/colorbar.py

```diff
@@ -238,6 +238,8 @@ def __init__(self, colorbar):
         super().__init__(nbins=nbins, steps=steps)
 
     def tick_values(self, vmin, vmax):
+        if vmin > vmax:
+            vmin, vmax = vmax, vmin
```

Comment:
This seems good.  `MaxNLocator` uses `mtransforms.nonsingular` to do the same thing.  Not sure its needed here...